openmdao/visualization/n2_viewer/n2_viewer.py

```diff
@@ -77,7 +94,7 @@ def _get_var_dict(system, typ, name):
             var_dict['value'] = type(meta['value']).__name__
     else:
         if meta['value'].size < _MAX_ARRAY_SIZE_FOR_REPR_VAL:
-            var_dict['value'] = meta['value']
+            var_dict['value'] = _convert_to_support_nans_in_json(meta['value'])
         else:
```

Comment:
I think you can probably add a test for this if calling `n2` on a prob with NaN formerly raised an exception.  Just do `n2(p, show_browser=False)` so that it doesn't open the browser or write a file.

Reply:
Great point. I had that in my head as a to do yesterday and then forgot today. Will add

Reply:
similar to other comment..
I think you could replace this single use function with:
`var_dict['value'] = _convert_nans_in_nested_list(val.tolist())`